### ISSUE TITLE
synthdetect: flow-matching TTS (Chatterbox) evades w2v2-aasist detection (S6 finding)

### ISSUE THREAD

**Issue by bengizmo** [bengizmo/voxint#252]
## Context

S6 composite corpus scoring (issue #144) reveals that **Chatterbox TTS** (AR autoregressive token prediction + conditional flow matching + HiFT source-filter vocoder) is near-undetectable by the production w2v2-aasist checkpoint:

| Generator | EER | AUC | Status |
|---|---|---|---|
| Piper (VITS) | 20.03% | 0.857 | Detected |
| ElevenLabs | 18.06% | 0.886 | Detected |
| Google Cloud TTS | 17.96% | 0.882 | Detected |
| **Chatterbox (AR+flow)** | **45.32%** | **0.558** | **Near chance** |
| ASVspoof DF anchor | 7.05% | 0.986 | Healthy |

59.4% of Chatterbox scores fall within the bona fide interquartile range. KS statistic (bona fide vs Chatterbox) = 0.14 (vs 0.66 for Piper). Even against AMI-only bona fide (cleanest comparison), EER = 40.3%.

## Root cause

A 4-model consult (codex + deepseek-v4-pro + grok-4.5 + kimi-k3) converged on:

1. **OOD (primary):** The production checkpoint (`LA_model.pth`) was trained on ASVspoof 2019 LA (6 attack systems). No flow-matching, no modern AR+flow TTS, no discrete speech tokens in the training data.
2. **Architecture mismatch:** Flow matching eliminates mel oversmoothing and prosodic collapse (acoustic-model artifacts). HiFT's source-filter vocoder eliminates GAN-specific phase/upsampling artifacts. Together they remove both artifact classes the detector learned.
3. **Voice-cloning reference transfer (secondary):** Chatterbox clones from each bona fide parent, inheriting channel/speaker cues that pull scores toward bona fide.

## Evidence this is closeable

DFADD (2024, arXiv 2409.08731) reports 44.21% average EER for ASVspoof-trained AASIST on unseen flow-matching TTS (VoiceBox 42.59%, VoiceFlow 50.41%), closely matching our 45.32%. Training AASIST on flow-matching examples reduced EER to ~23-25%.

## Impact on calibration

- A single threshold cannot serve both detectable (Piper, ElevenLabs, Google) and undetectable (Chatterbox-class) generator families
- Platt calibration will produce confident "bona fide" probabilities for Chatterbox-class TTS
- The shipped detector must report coverage limitations explicitly

## Proposed experiments (prioritized)

1. **Few-shot fine-tune probe:** Fine-tune on k={20, 100, 500} Chatterbox utterances from held-out speakers. If EER < 10% at k=500, coverage gap confirmed and fine-tuning is the fix.
2. **Class-transfer test:** Fine-tune on Chatterbox, test zero-shot on a different flow-matching TTS (F5-TTS, CosyVoice2). Transfer = learnable class signature; no transfer = per-generator whack-a-mole.
3. **Layer-wise XLS-R probe:** Linear probes per wav2vec2 layer. If no layer separates Chatterbox above chance, residual artifacts are genuinely minimal.
4. **Held-out reference speaker ablation:** Regenerate Chatterbox with speakers NOT in the bona fide pools to isolate the cloning confound.

## Related

- Part of #144 (M1 eval harness)
- Scoring results in `docs/gpu-contracts.md` S6 verdict block
- Also found: VoxConverse channel confound (separate from evasion, affects all generators)

## Decision needed

Whether to:
- (a) Proceed with S7 calibration acknowledging this limitation in the coverage statement
- (b) Run one or more of the proposed experiments first
- (c) Expand the corpus with more diverse reference conditions before calibrating

**Comment by bengizmo** [comment]
Synthdetect eval tooling extracted to private repo (PR #272). Public evidence summary retained in docs/gpu-contracts.md:

- Baseline w2v2-aasist: 37.77% EER on Chatterbox eval subset
- After M2 fine-tuning (#268): 25.90% EER (11.87 pp improvement)
- Other generators: Piper 8.04%, ElevenLabs 11.20%, Google 10.28%
- Progressive XLS-R unfreezing (Phase 3) available on 24 GB GPU if further improvement needed

Full eval protocols and dated verdicts now in the private repo.

**Comment by bengizmo** [comment]
## Scoped Plan: M3 Flow-Matching Detection Improvement

4-model consult (codex, deepseek-v4-pro, grok-4.5, kimi-k3) converged on the approach below. Full plan: `~/.claude/plans/zazzy-sleeping-diffie.md`.

### Root Cause

The 23pp dev-to-eval gap (2.47% dev vs 25.90% eval) points to **codec/channel overfitting** as the dominant factor, not insufficient backend capacity. Codec augmentation is the highest-ROI intervention.

### Targets (dual metrics)

| Gate | Chatterbox EER | TPR @ 1% FPR | Meaning |
|---|---|---|---|
| Interim shippable | ≤18% | ~30% | Worth shipping with defensive UX |
| **Primary ship gate** | **≤15%** | **~40%** | **Meaningful advisory signal** |
| Stretch (Phase 3) | ≤10% | ~60% | Genuinely useful |

All gates evaluated on **worst-of-3-seeds**. Non-regression: Piper/EL/Google ≤ M2+2pp, ASVspoof DF ≤ M2+4pp, bonafide FPR ≤5% AMI / ≤10% VC.

### Phases

**Phase 0: Contracts + Diagnostics** (2-3 days)
- Freeze eval protocol (per-generator + pooled EER, TPR@1%FPR, codec-stratified)
- Create locked challenge partition (speaker/prompt-disjoint, never consulted)
- XLS-R layer probes on Chatterbox subset (determines if front-end or backend is the bottleneck)
- Generate Matcha-TTS + F5-TTS eval slices before any training
- Pre-register all gates in gpu-contracts.md

**Phase 1: Codec Augmentation** (5-7 days, darkboi 3060s)
- Symmetric MP3/Opus/AAC augmentation on both bonafide and spoof (prevents codec-label shortcut)
- Feature-cached XLS-R, AASIST backend, 3 seeds
- Three ablation runs: `+codec`, `+generators`, `+both`
- Ship gate: Chatterbox EER ≤18%, ≥5pp gain, non-regression

**Phase 2: FM Generator Diversity** (7-10 days)
- Add F5-TTS to training (diverse zero-shot voices)
- Hold Matcha-TTS entirely out as unseen-FM canary
- Equal per-generator sampling across all spoof sources
- Ship gate: Chatterbox EER ≤15%, Matcha EER ≤25%, non-regression

**Phase 2.5: Layer Fusion** (conditional, 2-3 days)
- Only if P0 probes showed early XLS-R layers carry FM signal
- SUPERB-style learned weighted sum, frozen front-end

**Phase 3: Progressive Unfreezing** (conditional, thinktank 3090s)
- Entry: best frozen model still >15% Chatterbox EER
- Unfreeze last 2-4 XLS-R blocks, discriminative LRs, ASVspoof forgetting canary
- Ship gate: Chatterbox EER ≤10%, ≥2pp over best frozen

**Phase 4: Promotion** (per qualifying phase)
- Platt recalibration, 3 cold-process determinism trials, Docker rebuild, provenance update, UX copy

### Decision Tree

```
IF P1 ≤15%: ship; P2 optional
ELIF P1 ≤18% AND gain ≥5pp: ship interim; proceed P2
ELIF P1 gain <3pp: skip P2, go P3 (representation bottleneck)

IF P2 ≤15% AND Matcha ≤25%: ship; P3 optional
ELIF P2 >15% AND probes showed signal: try P2.5 first
ELIF P2 >15%: proceed P3

IF P3 fails: close with best model + honest UX; open WavLM as separate issue
```

### Out of Scope

WavLM Large swap, multi-SSL ensemble, Scalable AASIST rewrite, domain-adversarial training, meta-LoRA, non-advisory behavior, cloud GPU.

### Risks

| Risk | Mitigation |
|---|---|
| Codec-label shortcut | Symmetric augmentation on both sides |
| Augmented clips inflate CIs | Bootstrap by parent/speaker groups |
| Seed instability | Worst-of-3-seeds gating |
| Chatterbox memorization | Held-out Matcha canary; equal per-gen sampling |
| Scope creep | WavLM/architecture explicitly excluded |

### Immediate Next

1. Phase 0 pre-registration (eval protocol, locked challenge manifest)
2. Generate Matcha-TTS + F5-TTS eval slices
3. Layer probes on Chatterbox subset
4. Codec pilot on ~200 parents
